### PR TITLE
Generic/CamelCapsFunctionName: improve code coverage

### PR DESCRIPTION
**Pull request by rodrigoprimo**
# Description

This PR improves code coverage for the `Generic.NamingConventions.CamelCapsFunctionName` sniff and makes the following minor changes to the sniff code:

- Removes two incorrect code comments.
- Update a code comment to better describe what it does.
- Removes an unnecessary `return;` at the end of a method.

## Related issues/external references

Part of https://github.com/PHPCSStandards/PHP_CodeSniffer/issues/146

## Types of changes

- [ ] Bug fix _(non-breaking change which fixes an issue)_
- [ ] New feature _(non-breaking change which adds functionality)_
- [ ] Breaking change _(fix or feature that would cause existing functionality to change)_
    - [ ] This change is only breaking for integrators, not for external standards or end-users.
- [ ] Documentation improvement

## PR checklist

- [x] I have checked there is no other PR open for the same change.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md).
- [x] I grant the project the right to include and distribute the code under the BSD-3-Clause license (and I have the right to grant these rights).
- [ ] I have added tests to cover my changes.
- [x] I have verified that the code complies with the projects coding standards.
- [ ] \[Required for new sniffs\] I have added XML documentation for the sniff.

### CODE REVIEW

File: src/Standards/Generic/Sniffs/NamingConventions/CamelCapsFunctionNameSniff.php [hunk 1]
```diff
@@ -113,7 +113,6 @@ protected function processTokenWithinScope(File $phpcsFile, $stackPtr, $currScop
 
         $methodName = $phpcsFile->getDeclarationName($stackPtr);
         if ($methodName === null) {
-            // Ignore closures.
```
Comment (jrfnl) [inline src/Standards/Generic/Sniffs/NamingConventions/CamelCapsFunctionNameSniff.php:116]: Historical context: there was a time in which PHPCS didn't retokenize the `T_FUNCTION` token to `T_CLOSURE` for closures, so at that time, the comment was perfectly correct.
The comment just wasn't updated when the tokenizer change was made.

```diff
+            // Live coding or parse error. Bow out.
```

(here and below)

File: src/Standards/Generic/Sniffs/NamingConventions/CamelCapsFunctionNameSniff.php [hunk 2]
```diff
@@ -150,7 +149,7 @@ protected function processTokenWithinScope(File $phpcsFile, $stackPtr, $currScop
             return;
         }
 
-        // Ignore first underscore in methods prefixed with "_".
+        // Ignore leading underscores in the method name.
```
Comment (jrfnl) [inline src/Standards/Generic/Sniffs/NamingConventions/CamelCapsFunctionNameSniff.php:152]: Missing same fix on line 207.

File: src/Standards/Generic/Tests/NamingConventions/CamelCapsFunctionNameUnitTest.1.inc [hunk 3]
```diff
@@ -183,3 +183,22 @@ enum Suit: string implements Colorful, CardGame {
     public function parseMyDSN() {}
     public function get_some_value() {}
 }
+
+interface MyInterface {
+    function getSomeValue(); // Ok.
+    function get_some_value(); // Not ok.
```
Comment (jrfnl) [inline src/Standards/Generic/Tests/NamingConventions/CamelCapsFunctionNameUnitTest.1.inc:189]:
```suggestion
    public function getSomeValue();
    public function get_some_value();
```

File: src/Standards/Generic/Sniffs/NamingConventions/CamelCapsFunctionNameSniff.php (hunk 1 above)
Comment (rodrigoprimo) [reply src/Standards/Generic/Sniffs/NamingConventions/CamelCapsFunctionNameSniff.php:116]: Done

File: src/Standards/Generic/Sniffs/NamingConventions/CamelCapsFunctionNameSniff.php (hunk 2 above)
Comment (rodrigoprimo) [reply src/Standards/Generic/Sniffs/NamingConventions/CamelCapsFunctionNameSniff.php:153]: Done

**Review by jrfnl — APPROVED** [review]
@rodrigoprimo All good. Thanks for making those updates.

I'll rebase & squash the fixup commits before merging.

**Pull request merged**